src/metabase/api/action.clj

```diff
@@ -83,7 +85,15 @@
    template               [:maybe http-action-template]
    response_handle        [:maybe json-query-schema]
    error_handle           [:maybe json-query-schema]}
+
   (api/write-check Card model_id)
+  (when (and (nil? database_id)
+             (= "query" type))
+    (throw (ex-info (tru "Must provide a database_id for query actions")
```

Comment:
confirmation: we can't do anything for implicit or HTTP actions since they're not tied to a specific database?